### ISSUE TITLE
Bug: Role update does not return permission

### ISSUE THREAD

**Issue by ihmpavel** [strapi/strapi#13763]
### Required System information

- Node.js version: 16.15.1
- NPM version: 8.11.0
- Strapi version: 4.2.2 (does not work from 4.1.x)
- Database: Postgres
- Operating system: MacOS/UNIX (OS irrelevant)

### Describe the bug
On Roles/Permission page I check checkboxes (all) `Collection Types` and save, the items are saved to DB. The response from the server is `{"data":[]}` from `/admin/roles/3/permissions` endpoint. After page reload, nothing changes (there are still dots and permissions does not work.)

I tried deleting the role `delete from admin_permissions where id in (select permission_id from admin_permissions_role_links where role_id = 3);`, but after saving items are created in DB, but still same issue.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/42217494/178732927-e9bacf5b-f42e-45f6-8bf9-466ebdba84a2.png">

### Steps to reproduce the behavior

_Maybe, only applicable on my environment. (Does not depend on Strapi version, but database is maybe broken - Development/Production - same issue)_ 
1. Go to 'Roles -> Permissions'
2. Update permissions
3. Save - `{"data":[]}` is returned from server, rows in DB are created
4. Reload page - Permission still has dots

### Expected behavior

Saving Permissions should work as for other ones - `SuperAdmin` 

### Screenshots
Expected 
<img width="300" alt="image" src="https://user-images.githubusercontent.com/42217494/178734080-d3eb753d-486b-44e9-884a-0d08d3569563.png">

Current
<img width="300" alt="image" src="https://user-images.githubusercontent.com/42217494/178734140-c6853ec4-4934-4b7b-ac23-16cd0d4e48d6.png">

**Comment by derrickmehaffy** [comment]
Cannot reproduce on v4.2.3: https://www.loom.com/share/243e87ee183d4c0ca6e530538ea2de71

Gonna mark as closed for now until we get some new information on how to reproduce this.

**Comment by ihmpavel** [comment]
Hi, that means something is broken in my database.

Is there a way to delete the role and create a new one in the Community edition?

As I stated in the issue, I tried deleting all permission (`delete from admin_permissions where id in (select permission_id from admin_permissions_role_links where role_id = 3);`) and hoping for re-creating them by saving the role (which saved the permissions), but that did not help and the bug still exists.

The weirdest part for me is that it returns data on initial load but does not work while saving. Only on one specific role and not others.
